pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
`pgrep` would suffice here I think. You might want to anchor the regex, certainly at the end but perhaps at the beginning and with a full path too (to avoid matching e.g. `docker-containerd`).

Thinking about this though, why does the presence of either of this indicate that we are in a namespace? They are both certainly present in the root too. What this check is really doing is trying to have a heuristic guess at whether we are in the root pid namespace, it needs to be combined with something else to have a guess about being in the root mnt namespace or not (it's the mnt namespace which was confusing people). @justincormack's `readlink` suggestion is probably the way to go here.